packages/google_sign_in/google_sign_in_platform_interface/lib/src/method_channel_google_sign_in.dart

```diff
@@ -25,12 +25,29 @@ class MethodChannelGoogleSignIn extends GoogleSignInPlatform {
     SignInOption signInOption = SignInOption.standard,
     String? hostedDomain,
     String? clientId,
+  }) {
+    return init2(
+      scopes: scopes,
+      signInOption: signInOption,
+      hostedDomain: hostedDomain,
+      clientId: clientId,
+    );
+  }
+
+  @override
+  Future<void> init2({
```

Comment:
Because the fact that adding any new parameter counts as breaking change here, what do you think about making parameters into a class so we don't have to do breaking changes like this in the future? For example:
```dart
class GoogleSigInParams {
     List<String> scopes,
    SignInOption signInOption 
    String? hostedDomain,
    String? clientId,
    String? serverClientId,
    ...
}
```

cc @ditman

Reply:
Yes, this is a good point, and the migration for people would be super easy.

Reply:
This has been a pretty standard practice when we've been doing this; I'll add it to https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-platform-interface-method-parameters

Reply:
> the migration for people would be super easy

And since this is the platform interface, "people" is just us (and anyone making unendorsed implementations), not clients of the plugin itself.

Reply:
Done